### PR TITLE
Add mask for disabled moves, small changes to model and reward

### PR DESCRIPTION
**Pull request by ljones315**
Creates a mask within the team object that indicates which actions are valid for the given turn. This disallows the model from selecting invalid moves.

Also adjusts the reward function to include bonuses for winning battles or knocking out pokemon (the second of which may be less useful). But it also scales the weighting of damage (immediate) rewards against the weighting of winning (long term) rewards as the episodes progress.

The model also has a LSTM component now, which I hoped would help it begin to train more effectively. 

As of now these additions don't significantly impact WR when compared to the main branch, but they may be useful once we improve embeddings, increase model knowledge, etc.

### CODE REVIEW

File: trainer.py [hunk 1]
```diff
@@ -85,28 +105,29 @@ def train(model_actor, random_actor, gamma, num_episodes, optimizer):
...
             # (1) High loss values could mean prob_action is close to 0 often (poor model convergence?)
             # (2) Hacky fix for now is to add 1e-10 to prob_action to avoid taking log of 0.
             loss = -torch.log(prob_action + 1e-10) * current_discounted_return
 
-            total_loss += loss # negative loss values are due to negative rewards
+            total_loss = total_loss + loss  # negative loss values are due to negative rewards
 
         # This check is done as sometimes training samples are too few / invalid causing total_loss to be 0
-        if (total_loss != 0):
+        if (total_loss != 0 and not torch.isnan(total_loss)):
```
Comment (AdavyaB7) [inline trainer.py:124]: One thing I have noticed in other implementations of loss is taking the mean over all samples before backpropagating. I can test it out once you merge, or if you want to make this quick change now, feel free!

File: model/custom_pokemon_model.py [hunk 2]
```diff
@@ -15,15 +17,21 @@ def __init__(self):
 
         self.numerical_fc = nn.Linear(7 * 12, EMBEDDING_DIM)  # 7 stats for each pokemon on both teams
 
+        self.lstm = nn.LSTM(
```
Comment (AdavyaB7) [inline model/custom_pokemon_model.py:20]: Ah man, now I have to update the model diagram which took forever to draw 🫠

**Review by AdavyaB7 — APPROVED** [review]

File: model/custom_pokemon_model.py (hunk 2 above)
Comment (ljones315) [reply model/custom_pokemon_model.py:20]: Haha I can help with that if needed. We may end up altering it even more before we finish 👀

File: trainer.py (hunk 1 above)
Comment (ljones315) [reply trainer.py:124]: Got it, I can try adding that
